app/controllers/ems_configuration_controller.rb

```diff
@@ -0,0 +1,103 @@
+class EmsConfigurationController < ApplicationController
+  include Mixins::GenericListMixin
+  include Mixins::GenericShowMixin
+  include Mixins::GenericSessionMixin
+  include Mixins::BreadcrumbsMixin
+  include Mixins::GenericFormMixin
+  include Mixins::ManagerControllerMixin
+  include Mixins::FindRecord
+
+  before_action :check_privileges
+  before_action :get_session_data
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  def self.model
+    ManageIQ::Providers::ConfigurationManager
+  end
+
+  def self.table_name
+    @table_name ||= "ems_configuration"
+  end
+
+  def self.display_methods
+    %w[configuration_profiles configured_systems]
+  end
+
+  def button
+    @edit = session[:edit] # Restore @edit for adv search box
+    params[:display] = @display if %w[configured_systems].include?(@display) # Were we displaying nested list
+
+    # Handle Toolbar Policy Tag Button
+    @refresh_div = "main_div" # Default div for button.rjs to refresh
+    model = self.class.model
+    tag(model) if params[:pressed] == "#{params[:controller]}_tag"
+
+    if params[:pressed].starts_with?("configured_system_") # Handle buttons from sub-items screen
+      tag(ConfiguredSystem) if params[:pressed] == "configured_system_tag"
+      provision if params[:pressed] == "configured_system_provision"
+    end
+
+    case params[:pressed]
+    when 'ems_configuration_edit_provider'
+      edit
+    when 'ems_configuration_add_provider'
+      new
+    when "ems_configuration_refresh_provider"
+      refresh
+    when "ems_configuration_delete_provider"
+      delete
+    end
+
+    return if ["#{params[:controller]}_tag"].include?(params[:pressed]) && @flash_array.nil? # Tag screen showing
```

Comment:
should we move this line right after the `tag(mode)` line? as both happen under the same button..